### PR TITLE
Add GetAirbases API

### PR DESCRIPTION
**Pull request by rurounijones**
Add API to retrieve airbase information. This includes things like
carriers and FARPS.

### CODE REVIEW

File: lua/exporters/object.lua [hunk 1]
```diff
@@ -19,11 +19,11 @@ end
...
-    id = unit:getID()
-    name = unit:getName()
-    callsign = unit:getCallsign()
-    coalition = unit:getCoalition();
-    type = unit:getTypeName()
+    id = unit:getID(),
+    name = unit:getName(),
+    callsign = unit:getCallsign(),
+    coalition = unit:getCoalition(),
+    type = unit:getTypeName(),
```
Comment (rurounijones) [inline lua/exporters/object.lua:26]: Added the commas for consistency with usage elsewhere

File: protos/common.proto [hunk 2]
```diff
@@ -47,6 +53,13 @@ message Scenery {
 }
 
 message Airbase {
+	string id = 1;
```
Comment (rurounijones) [inline protos/common.proto:56]: This doesn't actually work yet. The rust serializer errors out expecting that Id *must* be present

`021-04-02 16:09:24.797 ERROR   SCRIPTING: [GRPC] Error retrieving next command: Failed to deserialize result for method getAirbases: deserialize error: missing field id`

 which seems to be a bug based on https://github.com/danburkert/prost/issues/430 and the comment

> As the README notes, proto3 message fields without a modifier are be wrapped in an Option. The new proto3 optional modifier is for scalar fields, I'm kind of surprised protoc even lets you apply it to a message field.

As a test I did try to apply `optional` to this field but, as expected, this fails.

`protoc failed: common.proto:56:18: Explicit \'optional\' labels are disallowed in the Proto3 syntax. To define \'optional\' fields in Proto3, simply remove the \'optional\' label, as fields are \'optional\' by default`

**Review by rurounijones — COMMENTED** [review]
This is not ready to merge yet until the `optional` / not `optional` thing is figured out. See review comments for details

File: src/rpc.rs [hunk 3]
```diff
@@ -82,6 +82,18 @@ impl Mission for RPC {
...
 
+    async fn get_airbases(
+        &self,
+        request: Request<GetAirbasesRequest>,
+    ) -> Result<Response<GetAirbasesResponse>, Status> {
+        let res: GetAirbasesResponse = self
+            .ipc
+            .request("getAirbases", Some(request.into_inner()))
+            .await
+            .map_err(|err| Status::internal(err.to_string()))?;
```
Comment (rkusa) [inline src/rpc.rs:93]: I've simplified the API here part a bit, so we can shorten it to the following:

```suggestion
        let res: GetAirbasesResponse = self.request("getAirbases", request).await?;
```

Comment (rkusa) [reply src/rpc.rs]: I made the change, I hope you don't mind

**Review by rkusa — APPROVED** [review]

File: src/rpc.rs (hunk 3 above)
Comment (rurounijones) [reply src/rpc.rs]: Awesome, thanks!

**Comment by rurounijones** [comment]
Thanks for investigating and fixing.

I need to do some more testing with the various things that qualify as `airfields` (Aircraft carriers, helo carriers, FARPS, oilrigs etc) and will merge if that is successful.

**Comment by rurounijones** [comment]
Fully implemented the API allowing for optional specifying of the coalition Id. I am going for optional here since it makes more sense than adding another enum value for `all` which could bite us later.

**Pull request merged**